File: dispatch/src/main/java/com/oman/dispatch/normal/TestActivity.java
package com.oman.dispatch.normal;

import android.os.Bundle;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;

import androidx.appcompat.app.AppCompatActivity;

import com.oman.dispatch.R;

import static com.oman.dispatch.normal.Utils.getAction;

public class TestActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_test);

        View rootView = findViewById(R.id.rootView);
        final Button button = findViewById(R.id.buttonView);
        final ImageView imageView = findViewById(R.id.imageView);

        rootView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.i("normally", "father onclick");
            }
        });

        rootView.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                Log.i("normally", "father onTouch" + getAction(event.getAction()));
                return false;
            }
        });

        button.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.e("normally", "button onClick");
            }
        });

        button.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                Log.e("normally", "button onTouch: " + getAction(event.getAction()) + "--clickable:" + button.isClickable());

                return true;
            }
        });

        imageView.setOnTouchListener(new View.OnTouchListener() {
            @Override
            public boolean onTouch(View v, MotionEvent event) {
                Log.e("normally", "image onTouch: " + getAction(event.getAction()) + "--clickable:" + imageView.isClickable());
                return true;
            }
        });
        imageView.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Log.e("normally", "image onClick");
            }
        });

    }
}

File: app/src/main/java/com/oman/forward/TestActivity.kt
package com.oman.forward

import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.oman.forward.adapter.TestAdapter

class TestActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_test)
        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)
        toolbar.title = "Home"
        setSupportActionBar(toolbar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        val categories = resources.getStringArray(R.array.categories).toList()
        val names = resources.getStringArray(R.array.names).toList()
        val namesAll = listOf(names, names, names, names, names)
        val appList = findViewById<RecyclerView>(R.id.appList)
        val adapter = TestAdapter(this, namesAll, categories)
        val manager = LinearLayoutManager(this)
        appList.layoutManager = manager
        appList.adapter = adapter
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            finish()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
